### PR TITLE
Configure gulp

### PR DESCRIPTION
**Pull request by m-recz**
Add a task in gulp to convert css into scss files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.vscode
+css/
+node_modules/
```
Comment (Bosmanfrx) [inline .gitignore]: newline

File: assets/images/header-bg-mobile.svg [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+ <g>
+  <title>background</title>
+  <rect fill="none" id="canvas_background" height="217" width="1242" y="-1" x="-1"/>
+ </g>
+ <g>
+  <title>Layer 1</title>
+  <path stroke="null" id="svg_1" fill="#0D4C5E" d="m1283.6,89.750787c-86.93,144.065091 -1021.559,-33.119436 -1250.2092,107.824463c-228.6498,140.945194 -14.4346,-232.016668 -14.4346,-232.016668l1250.2138,-107.824581c0,0 101.36,87.951812 14.43,232.016785z"/>
+ </g>
+</svg>
```
Comment (Bosmanfrx) [inline assets/images/header-bg-mobile.svg]: newline

File: assets/images/header-bg.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg viewBox="0 0 1240 189" fill="none" xmlns="http://www.w3.org/2000/svg">
+                <path d="M1283.6 54.7912C1196.67 177.149 262.041 26.6621 33.3908 146.369C-195.259 266.077 18.9562 -50.6881 18.9562 -50.6881L1269.17 -142.266C1269.17 -142.266 1370.53 -67.5665 1283.6 54.7912Z" fill="#0D4C5E"/>
+                </svg>
```
Comment (Bosmanfrx) [inline assets/images/header-bg.svg]: newline

File: assets/scss/_base.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+* {
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+}
```
Comment (Bosmanfrx) [inline assets/scss/_base.scss]: newline

File: assets/scss/_base.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+* {
```
Comment (Bosmanfrx) [inline assets/scss/_base.scss]: actually, we could go with `*, *::after, *::before` - it will apply these styles to pseudo-elements as well ;)

File: assets/scss/_header.scss [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    background-size: 100% auto;
+    background-repeat: no-repeat;
+
+    @include media-queries('tablet') {
+        background-image: url('../images/header-bg.svg');
+    }
+}
+
+.nav__list {
+    min-height: 20vw;
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: We set here height by device width. Is it desired solution? I think, that on very large screens that header will be very tall, likewise on small devices - it will be small. You could use set `px` height, or even better - don't set height at all. Let content decide about height

File: assets/scss/_header.scss [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    display: flex;
+    align-items: center;
+    list-style: none;
+    padding: 0 5vw 5.5vw;
+
+    @include media-queries('tablet') {
+        min-height: 15vw;
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: ditto

File: assets/scss/_header.scss [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        background-image: url('../images/header-bg.svg');
+    }
+}
+
+.nav__list {
+    min-height: 20vw;
+    display: flex;
+    align-items: center;
+    list-style: none;
+    padding: 0 5vw 5.5vw;
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: Padding will look little different on different devices. It's the best to set some constant value (like `rem`). It will only change if user has bigger font size then

File: assets/scss/_header.scss [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    text-decoration: none;
+}
+
+.nav__item:first-child {
+    flex-grow: 1;
+}
+
+.nav__logo {
+    font-family: 'Roboto', sans-serif;
+    font-size: 30px;
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss:32]: Could we use rem here?

File: assets/scss/_header.scss [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    text-decoration: none;
+
+    @include media-queries('tablet') {
+        font-size: 28px;
+    }
+}
+
+img {
+    max-width: 100%;
+}
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: newline

File: assets/scss/_header.scss [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    color: white;
+    text-decoration: none;
+
+    @include media-queries('tablet') {
+        font-size: 28px;
+    }
+}
+
+img {
+    max-width: 100%;
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: `height: auto`

File: assets/scss/_header.scss [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    font-size: 18px;
+    color: white;
+    text-decoration: none;
+
+    @include media-queries('tablet') {
+        font-size: 28px;
+    }
+}
+
+img {
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: from what I see it will apply to every element on site. Maybe it belongs somewhere in `main` or `_base`

File: assets/scss/_maps.scss [hunk 13]
```diff
@@ -0,0 +1,2 @@
+$breakpoints: (tablet: (min-width: 768px),
+    desktop: (min-width: 1240px))
```
Comment (Bosmanfrx) [inline assets/scss/_maps.scss]: newline

File: assets/scss/_mixins.scss [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+    @if($size) {
+        @media #{$size} {
+            @content;
+        }
+    }
+
+    @else {
+        @error '"#{$breakpoint}" - not recognized size';
+    }
+}
```
Comment (Bosmanfrx) [inline assets/scss/_mixins.scss]: newline

File: assets/scss/style.scss [hunk 15]
```diff
@@ -0,0 +1,4 @@
+@import 'maps';
+@import 'mixins';
+@import 'base';
+@import 'header';
```
Comment (Bosmanfrx) [inline assets/scss/style.scss]: newline

File: gulpfile.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+
+gulp.task('sass', function () {
+    return gulp.src('assets/scss/*.scss')
+        .pipe(sass().on('error', sass.logError))
+        .pipe(gulp.dest('assets/css/'));
+});
+
+gulp.task('sass:watch', function () {
+    gulp.watch('assets/scss/*.scss', ['sass']);
+});
```
Comment (Bosmanfrx) [inline gulpfile.js]: newline

File: gulpfile.js [hunk 17]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+var gulp = require('gulp');
+var sass = require('gulp-sass');
+
+sass.compiler = require('node-sass');
+
+gulp.task('sass', function () {
+    return gulp.src('assets/scss/*.scss')
```
Comment (Bosmanfrx) [inline gulpfile.js]: I think it will produce many css files here. Maybe we don't need them, since we import `style.css` only. We could cut it down to `assets/scss/style.scss`

File: package.json [hunk 18]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "alinka-website",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (Bosmanfrx) [inline package.json]: It can be safely removed

File: package.json [hunk 19]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "alinka-website",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
```
Comment (Bosmanfrx) [inline package.json:4]: We could set script for running gulp here. We could specify scripts like `"watch": "gulp sass:watch"` etc. It will not require knowledge of npx to run this project

File: package.json [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/CodeForPoznan/alinka-website.git"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (Bosmanfrx) [inline package.json]: Is this license we want? Git repository specifies `MIT`

File: package.json [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/CodeForPoznan/alinka-website.git"
+  },
+  "author": "",
```
Comment (Bosmanfrx) [inline package.json]: We could set it to CodeForPoznan like here: https://github.com/CodeForPoznan/alinka-electron/pull/100/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R5

**Review by Bosmanfrx — CHANGES_REQUESTED** [review]

File: README.md [hunk 22]
```diff
@@ -0,0 +1,4 @@
+# How to start the project
+- (If you don't already have it, install node.js)
+- use `npm install` command to install the project
+- run `npx gulp sass` command to compile scss to css
```
Comment (Bosmanfrx) [inline README.md]: see package.json comment

File: assets/scss/_header.scss [hunk 23]
```diff
@@ -0,0 +1,57 @@
+.header {
+    background-image: url('../images/header-bg-mobile.svg');
+    background-size: 100% auto;
+    background-repeat: no-repeat;
+
+    @include media-queries('tablet') {
+        background-image: url('../images/header-bg.svg');
+    }
```
Comment (agatakozinska) [inline assets/scss/_header.scss]: you're using SVG so there is no need for second background, just manipulate its size ;)

File: assets/images/header-bg.svg (hunk 3 above)
Comment (agatakozinska) [inline assets/images/header-bg.svg]: please use prettier or another VS Code extension to get rid of those whitespaces :)

File: assets/scss/_header.scss (hunk 8 above)
Comment (agatakozinska) [reply assets/scss/_header.scss]: I agree, usually, vw is used for width but for padding and margin, it can be risky because it will be changing for all different devices

File: assets/scss/_header.scss [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+.nav__item:first-child {
+    flex-grow: 1;
+}
+
+.nav__logo {
+    font-family: 'Roboto', sans-serif;
+    font-size: 30px;
+    color: white;
```
Comment (agatakozinska) [inline assets/scss/_header.scss]: normally you shouldn't use `white`, it's better to have a file with variables and set to this variable #fff (not as a word but as a HEX)

File: assets/scss/style.scss (hunk 15 above)
Comment (agatakozinska) [inline assets/scss/style.scss]: imports and files should have _ 
read about underscore convention of naming scss files: 
https://stackoverflow.com/questions/34889962/why-put-in-front-of-the-file-name-or-in-scss-css
section partials: https://kolosek.com/sass-import/

File: gulpfile.js (hunk 17 above)
Comment (agatakozinska) [reply gulpfile.js]: > I think it will produce many css files here. Maybe we don't need them, since we import `style.css` only. We could cut it down to `assets/scss/style.scss`

Not really, if they use scss partials (I mentioned about it in review), it will create just one css file while compiling. Usually in gulp task: styles (here sass), you also put autoprefixer, plumber, cssnano, etc. and then create gulp.watch to watch this task, so I think it's necessary to see changes in all files and in which file problem occurred, otherwise you will see you have a problem in style.scss. But it depends on a configuration, right now it's a simple one.

**Review by agatakozinska — CHANGES_REQUESTED** [review]

File: assets/scss/_header.scss (hunk 6 above)
Comment (m-recz) [reply assets/scss/_header.scss]: The minimum height is set to prevent cutting of the background. The vw units are used to keep the proportions of the background. To keep the ratio, the height of it depends on the width. As we discussed, to prevent from having a problem on very large screens, I will add a new breakpoint for large screens and set max values for the page.

File: assets/scss/_header.scss [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+  background-size: 100% auto;
+  background-repeat: no-repeat;
+}
+
+.header__nav {
+  min-height: 13vw;
+}
+
+.nav__list {
+  // min-height: 13vw;
```
Comment (Bosmanfrx) [inline assets/scss/_header.scss]: left out comment :P

File: assets/scss/_variables.scss [hunk 26]
```diff
@@ -0,0 +1 @@
+$font-light: #fff;
```
Comment (Bosmanfrx) [inline assets/scss/_variables.scss]: newline

**Review by Bosmanfrx — CHANGES_REQUESTED** [review]

File: gulpfile.js (hunk 17 above)
Comment (m-recz) [reply gulpfile.js]: We do use scss partials and the the code is compiling into one css file. I also added in gulp-task to create the css sourcemap to easier find the problems.

File: assets/scss/style.scss (hunk 15 above)
Comment (m-recz) [reply assets/scss/style.scss]: The files already have _ and they are compiled into a single css file. From what I read, the _ is not required at import. The underscore is only necessary at import when you have e.g. header.scss and _header.scss, otherwise, the following import will give exactly the same results:
```
@import 'header';
@import '_header';
```

**Review by Bosmanfrx — APPROVED** [review]

File: assets/scss/_header.scss [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+      padding: 1.2rem 3rem 1.7rem;
+    }
+  }
+
+  &__item {
+    padding: 0.2rem 1rem 2rem;
+    @include media-queries("tablet") {
+      padding: 0.2rem 3rem 2rem;
+    }
+    text-decoration: none;
```
Comment (agatakozinska) [inline assets/scss/_header.scss:23]: it's not readable, at first should be all styles for mobile, so just after `padding` ->  `text-decoration` then media-queries

File: index.html [hunk 28]
```diff
@@ -1,102 +1,156 @@
...
+      type="text/css"
+    />
+    <link
+      href="https://fonts.googleapis.com/css?family=Dancing+Script|Montserrat:400,500,700|Oswald:200|Playfair+Display|Roboto|Roboto+Condensed&display=swap&subset=latin-ext"
+      rel="stylesheet"
+    />
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
     <script src=".config/config.js"></script>
     <script src=".config/local_config.js"></script>
     <script src="assets/js/send_form_to_google.js"></script>
```
Comment (agatakozinska) [inline index.html:20]: do you use any of these?

**Review by agatakozinska — APPROVED** [review]
just small things that you can change, but you have also my approve ;d

Edit:
maybe you could add also browsersync to run the project on a local server? right now when you do step by step `How to start the project` you can't actually see this website until you run some live server extension
